Makefile

```diff
@@ -123,43 +134,36 @@ $(PROTOC_GEN_DEEPCOPY):
 $(KIND):
 	@GOBIN=$(LOCALBIN) go install -mod=readonly sigs.k8s.io/kind@v0.26.0
 
+$(CONTROLLER_GEN):
+	GOBIN=$(LOCALBIN) go install -mod=readonly sigs.k8s.io/controller-tools/cmd/controller-gen@$(CONTROLLER_TOOLS_VERSION)
+
 .PHONY: clean
-clean: 
+clean:
 	@rm -rf $(LOCALBIN) $(PROJECT_DIR)/$(OUT_DIR)
 
 ##@ Code Gen
 
 $(PROTOBUF_GEN): $(PROTOBUF_API_SRC) $(PROTOC) $(PROTOC_GEN_GO) $(PROTOC_GEN_GRPC) $(PROTOC_GEN_DEEPCOPY) ## Generates Go files from protobuf-based API files
 	@PATH=$(LOCALBIN):$$PATH $(PROTOC) --proto_path=$(PROTOBUF_API_DIR) --go_out=$(API_GEN_DIR) --go-grpc_out=$(API_GEN_DIR) --golang-deepcopy_out=:$(API_GEN_DIR) $(PROTOBUF_API_DIR)/**/*.proto
 
+$(DEEP_COPY_GEN): $(DEEP_COPY_SRC) ## Generate code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations.
+	$(CONTROLLER_GEN) object:headerFile=$(LICENSE_FILE) paths="./..."
+
+$(CRD_GEN): $(CRD_SRC) ## Generate CustomResourceDefinition objects.
+	$(CONTROLLER_GEN) crd paths="./..." output:crd:artifacts:config=chart/crds
+
 .PHONY: fix-imports
 fix-imports: $(GOIMPORTS) ## Fixes imports
 	$(GOIMPORTS) -local "github.com/openshift-service-mesh/federation" -w $(PROJECT_DIR)/
 
-LICENSE_FILE := /tmp/license.txt
+LICENSE_FILE := $(PROJECT_DIR)/hack/boilerplate.go.txt
 GO_FILES := $(shell find $(PROJECT_DIR)/ -name '*.go')
 
 .PHONY: add-license
 add-license: ## Adds license to all Golang files
```

Comment:
Do we need it if we have license in the boilerplate file?

Reply:
I don't think it covers all sources, only those which are generated (`object:headerFile`)